### DISCUSSION TITLE
Allow direct array indexing of array member field in a #raw_union struct

### DISCUSSION THREAD

**Discussion by rowbyte** [odin-lang/Odin#3256] [Proposals]
Right now, one cannot override the swizzle fields/set custom swizzle field order or names in an array type (which is understandable). A decent workaround seems to be creating a `#raw_union` with the struct and an array field. But direct array indexing of this struct isn't supported.

```
AHLS :: struct #raw_union {
      using _  : struct {
                 alpha, hue, lightness,saturation : f32
      }

      using _ : [4] f32;
 
}
```

**Describe the solution you'd like**
Allow direct array indexing of raw-unions, if there is only one array field member inside the raw union struct.

**Additional context**
None.

**Comment by shayded-exe** [comment]
IMHO just adding custom swizzle names would be better, but this would also be nice.

**Reply by gingerBill** [reply]
NEVER going to happen.

**Comment by MaliusArth** [comment]
I think it would be more precise to say: export array indexing & swizzling syntax when `using` arrays to export them into parent scope

**Comment by shayded-exe** [comment]
Slight issue, I'm getting an error when trying to use `b` as a name because the default swizzling already defines it.

`'b' is already declared in 'struct #raw_union {using _: struct {h, s, b: f32}, using _: [3]f32}'(checker)`

```odin
HSB :: struct #raw_union {
	using _: struct {
		h, s, b: f32,
	},
	using _: [3]f32,
}
```

**Reply by MaliusArth** [reply]
I guess it would make sense if odin dropped the builtin aliases & swizzles (or at least made them optional) and fully supported custom aliasing & swizzling.

Starting by omitting auto array aliases & swizzles inside `#raw_union`s

**Comment by gingerBill** [comment]
Not going to happen. I understand why you'd want this and we experimented with it a while ago too, but it screws up a lot of rules too with `using`.

And in the grand scheme of things, this isn't that bad:
```odin
HSB :: struct #raw_union {
	using hsb: struct {h, s, b: f32},
	e: [3]f32,
}

col: HSB
col.s = 0.5
col.e[1] = 0.5 // an extra 2 characters to type.
```
